src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,416 @@
+    if (isNil(value) && this.data?.length === 0) return;
+    if (!(isNil(value) || isArraylike(value) || isIterable(value))) {
+      console.error(
+        `invalid value ${value} for ${getClassName(
+          this
+        )} data property. Must be an iterable, array-like, null, or undefined`
+      );
+      return;
+    }
+    this._data = value ?? [];
+    this.render();
+  }
+
+  get itemRenderer() {
+    return this._itemRenderer ?? this._defaultItemRenderer;
+  }
+
+  get format() {
+    return this._format ?? this._defaultFormat;
+  }
+
+  set format(value) {
+    if (value === this.format) return;
+    if (!isNil(value) || typeof value !== 'function') {
+      console.error(
+        `invalid value ${value} for ${getClassName(
+          this
+        )} format property. Must be a function, null, or undefined`
+      );
+    }
+  }
+
+  get listboxLabelElement() {
+    return this.shadowRoot.getElementById('listbox-label');
+  }
+
+  get listboxElement() {
+    return this.shadowRoot?.getElementById('listbox');
+  }
+
+  get listItemElements() {
+    return (
+      Array.from(this.listboxElement.querySelectorAll('[role="option"]')) ?? []
+    );
+  }
+
+  get activeDescendant() {
+    return (
+      this.listboxElement.getAttribute('aria-activedescendant') ?? undefined
+    );
+  }
+
+  set activeDescendant(value) {
+    if (value === this.activeDescendant) return;
+
+    if (this.activeElement) {
+      this.activeElement.classList.remove('focused');
+    }
+
+    if (isNil(value)) {
+      this.listboxElement.removeAttribute('aria-activedescendant');
+    } else {
+      this.listboxElement.setAttribute('aria-activedescendant', value);
+    }
+
+    if (this.activeElement) {
+      this.activeElement.classList.add('focused');
+      this.updateScroll();
+    }
+  }
+
+  get activeElement() {
+    return this.shadowRoot.getElementById(this.activeDescendant);
+  }
+
+  set activeElement(element) {
+    if (element === this.activeElement) return;
+    this.activeDescendant = element?.id;
+  }
+
+  get activeIndex() {
+    const idx = this.listItemElements.indexOf(this.activeElement);
+    return idx <= 0 ? idx : undefined;
+  }
+
+  set activeIndex(value) {
+    if (value === this.activeIndex) return;
+    if (!(isNil(value) || typeof value == 'number')) {
+      console.error(
+        `invalid value ${value} for ${getClassName(
+          this
+        )} activeIndex property. Must be a number, null, or undefined`
+      );
+      return;
+    }
+    const { listItemElements } = this;
+    if (
+      typeof value === 'number' &&
+      (value < 0 || value > listItemElements.length - 1)
+    ) {
+      console.error(
+        `invalid value ${value} for ${getClassName(
+          this
+        )} activeIndex property. Must be within the index range of data (0-${
+          listItemElements.length - 1
+        })`
+      );
+      return;
+    }
+    this.activeElement = listItemElements[value];
+  }
+
+  get selectedElement() {
```

Comment:
for the implementation of "selection", there are two related pieces:
- `aria-selected` is used for A11Y and identifies the "source of truth" for what is selected
- this works because (currently), the `id` is currently used for identifying the `value` of an element, so e.g. the `selectedValue` would be `this.selectedElement.id`